igcollect/ping.py

```diff
@@ -0,0 +1,179 @@
+    parser.add_argument('hosts', nargs='+',
+                        help='the hosts to ping')
+    parser.add_argument('-c', '--count', type=int, default=20,
+                        choices=range(1,60),
+                        help='how many times the script will try to ping '
+                             '(number has to be bigger '
+                             'than 1 and smaller than 60)')
+    parser.add_argument('-p', '--prefix',
+                        default='ping',
+                        help='the path to the value in Graphite ')
+    parser.add_argument('-t', '--timeout', type=int, default=500,
+                        help='Initial target timeout in milliseconds. '
+                             'the amount of time that program '
+                             'waits for a response')
+    parser.add_argument('-d', '--delay', type=int, default=1800,
+                        choices=range(0,5000),
+                        help='time to wait between a series of pings in '
+                             'milliseconds')
+    return parser.parse_args()
+
+
+def main():
+    """ main function to control Multitasking the collection of the pings """
+
+    args = parse_args()
+
+    chunked_hosts = [args.hosts[i:i + 10]
+                     for i in range(0, len(args.hosts), 10)]
+
+    data = [(args.prefix, chunk, args.count, args.timeout, args.delay)
+            for chunk in chunked_hosts]
+
+    with Pool(cpu_count() * 2) as p:
+        p.starmap(check_pings, data)
+
+
+def check_pings(prefix, hosts, count, timeout, delay):
+    """ checks the pings and prints them
+
+        Parameters:
+            prefix (str): the prefix for the output for graphite
+            hosts (list(str)):  all the hosts that get pinged
+            count (int): how many times a individual host get pinged
+            timeout (int): the time till the timeout of fping
+            delay (int): time to wait between a series of pings
+
+    """
+
+    values = pings(hosts, count, timeout, delay)
+
+    timestamp = str(int(time.time()))
+
+    for data in values:
+        print_ping(prefix, data, timestamp)
+
+
+def pings(hosts, count, timeout, delay):
+    """ pings the host and calculates all data needed
+
+        Parameters:
+            hosts (list(str): all the hosts that get pinged
+            count (int): how many times a individual host get pinged
+            timeout (int): the time till the timeout of fping
+            delay (int): time to wait between a series of pings
+
+        Returns:
+            values (list(dict(float))): all data gathered for all pinged hosts
+    """
+
+    hosts_string = ' '.join(hosts)
+    cmd = 'fping -B1 -q -C {} -p {} -t {} {}'
+    cmd = cmd.format(count, delay, timeout, hosts_string)
+    # output:
+    # example.com : 105.94 - 104.78
+    # Showing times for each ping. A ping timeout will be shown as '-'.
+    output = subprocess.getoutput(cmd).split('\n')
```

Comment:
Please switch to `subprocess.check_output()`, so that a shell is not spawned. Running things in shell is a bad security practice.